scripts/core/memory.yml

```diff
@@ -434,28 +434,59 @@ params:
       desc: |
             [out] a pointer to the native handle of the mem.
 --- #--------------------------------------------------------------------------
+type: struct
+desc: "Native memory object creation properties"
+class: $xMem
+name: $x_mem_native_properties_t
+base: $x_base_properties_t
+members:
+    - type: bool
+      name: isNativeHandleOwned
+      desc: >
+        [in] Indicates UR owns the native handle or if it came from an
+        interoperability operation in the application that asked to not
+        transfer the ownership to the unified-runtime.
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Native memory object creation properties"
+details:
+    - Specify these properties in $xMemCreateWithNativeHandle via
+      $x_mem_native_properties_t as part of a `pNext` chain.
+class: $xMem
+name: $x_mem_image_native_properties_t
+base: $x_base_properties_t
+members:
+    - type: "const $x_image_format_t*"
+      name: pImageFormat
+      desc: "[in] pointer to image format specification"
+    - type: "const $x_image_desc_t*"
+      name: pImageDesc
+      desc: "[in] pointer to image description"
+--- #--------------------------------------------------------------------------
 type: function
-desc: "Create runtime mem object from native mem handle."
+desc: "Create runtime memory object from native memory handle."
 class: $xMem
 name: CreateWithNativeHandle
 decl: static
 ordinal: "0"
 details:
-    - "Creates runtime mem handle from native driver mem handle."
-    - "The application may call this function from simultaneous threads for the same context."
-    - "The implementation of this function should be thread-safe."
+    - See also $x_mem_image_native_properties_t.
+    - The application may call this function from simultaneous threads for the
+      same context.
+    - The implementation of this function should be thread-safe.
 params:
     - type: $x_native_handle_t
       name: hNativeMem
-      desc: |
-            [in] the native handle of the mem.
+      desc: "[in] the native handle to the memory."
     - type: $x_context_handle_t
       name: hContext
-      desc: "[in] handle of the context object"
+      desc: "[in] handle of the context object."
+    - type: const $x_mem_native_properties_t*
+      name: pProperties
+      desc: "[in][optional] pointer to native memory creation properties."
     - type: $x_mem_handle_t*
       name: phMem
-      desc: |
-            [out] pointer to the handle of the mem object created.
+      desc: "[out] pointer to handle of image object created."
```

Comment:
mem object better, no? since it wont be always an image.

Reply:
Yes, this was a mistake from a previous iteration with two separate entry points. I'll revert to the old wording.

Reply:
Fixed.